src/image/components/runtime.tsx

```diff
@@ -33,17 +48,88 @@ export const Runtime: React.FC<IProps> = ({ authoredState, report }) => {
   const getOriginalImageSize = (e: any) => {
     if (e.target && e.target.naturalWidth && e.target.naturalHeight) {
       imageSize.current = { width: e.target.naturalWidth, height: e.target.naturalHeight };
-      const aspectRatio = e.target.naturalWidth / e.target.naturalHeight;
-      setSupportedFeatures({
-        interactiveState: true,
-        authoredState: true,
-        aspectRatio
-      });
+
+      if (authoredState.scaling === "fitWidth") {
+        if (!baseContainer.current) {
+          // FIMXE:
+          // If the image loads before the baseContainer is initialized this will
+          // default to LARA default aspect ratio of 1.3
+          // That will result in a clipped image in most cases.
+          // A better solution would be to delay the execution of this method
+          // until both the image is loaded and the baseContainer is available
+          console.error("image interactive baseContainer not available. aspectRatio will be wrong");
+          return;
+        }
+        // We use the baseContainer height to capture all of the spacing
+        // in the container. We can't use the image's aspect ratio
+        // directly because there is various bits of padding around the container.
+        const baseContainerHeight = baseContainer.current.scrollHeight;
+
+        // HACK: the aspectRatio is dependent on the iframe's width. When the iframe
+        // is more narrow, then the caption and magnifying glass will
+        // take up a larger percentage of the height. Additionally the caption
+        // can start wrapping so its actual pixel height will get larger.
+        // It is possible for the interactive to send a new aspectRatio
+        // whenever its width changes. However this can cause a race or infinite
+        // loop:
+        //   If the interactive is in the interactive box and the user resizes
+        //   the window then the width of the interactive will change. This would
+        //   trigger a re-calcuation of the aspectRatio in the interactive.
+        //   When the interactive sends this new aspectRaio to LARA,
+        //   LARA will re-calcuate the width and height of the interactive again.
+        //   LARA's calculation of the width will change if the
+        //   users browser is short so maintainig the aspect ratio means limiting
+        //   the width.  This width change will trigger the interactive to send a
+        //   new aspect ratio. This will go on and on.
```

Comment:
Have you tried this approach in practice?
I agree that it's worth it, I'm just curious. I could imagine the loop should automatically stop itself. And/or a reasonable stop condition could be not to send a new aspectRatio to LARA if it's very close to the current one.

Reply:
I haven't tried it with this interactive. 

But we did try something like this before we had the setHeight api and the interactive continued to jitter around. Partly because the height calculation resets it to 100% width before each calculation. So it isn't the case that we are getting closer and closer to some final state. Instead from the interactives point of view it is jumping to a large width, then jumping back to a close to the fine width, then jumping to a large width, ...